compiler/rustc_mir_build/src/build/mod.rs

```diff
@@ -719,8 +730,9 @@ fn construct_error<'a, 'tcx>(hir: Cx<'a, 'tcx>, body_id: hir::BodyId) -> Body<'t
     }
     builder.cfg.terminate(START_BLOCK, source_info, TerminatorKind::Unreachable);
     let mut body = builder.finish();
-    if tcx.hir().body(body_id).generator_kind.is_some() {
-        body.yield_ty = Some(ty);
+    if generator_kind.is_some() {
+        // this should fail in at least one place
```

Comment:
What do you mean?

Reply:
Bah, leftover for when I was debugging the first failures